repos/system_upgrade/common/actors/scansubscriptionmanagerinfo/tests/test_scansubscriptionmanagementinfo.py

```diff
@@ -0,0 +1,349 @@
+
+    assert not produced_message.sca_detected, 'The actor detected SCA on a on-SCA system.'
+
+
+def test_scansubscriptionmanagementinfo_is_manifest_sca_on_sca_system(monkeypatch, common_monkeypatched_handles):
+    """Tests whether the actor obtains the SCA information from SCA system correctly."""
+
+    _, actor_products, external_command_runner = common_monkeypatched_handles
+
+    external_command_runner.add_mocked_command_with_stdout(RHSM_STATUS, RHSM_STATUS_OUTPUT_SATELLITE)
+
+    scanrhsm.scan()
+
+    assert actor_products.model_instances, 'The actor did not produce any messages.'
+    produced_message = actor_products.model_instances[0]
+
+    assert produced_message.sca_detected, 'The actor failed to detected SCA on a SCA system.'
+
+
+def test_scansubscriptionmanagementinfo_get_enabled_repo_ids(monkeypatch, common_monkeypatched_handles):
+    """Tests whether the actor retrieves information about enabled repositories."""
+
+    _, actor_products, external_command_runner = common_monkeypatched_handles
+
+    # Prepare the (realistic) RHSM output
+    rhsm_list_enabled_output = '''
+    +----------------------------------------------------------+
+       Available Repositories in /etc/yum.repos.d/redhat.repo
+    +----------------------------------------------------------+
+    '''
+
+    for enabled_repository in RHSM_ENABLED_REPOS:
+        rhsm_output_fragment = 'Repo ID: {0}\n'.format(enabled_repository.id)
+        rhsm_output_fragment += 'Repo Name: {0}\n'.format(enabled_repository.name)
+        rhsm_output_fragment += 'Repo URL: {0}\n'.format(enabled_repository.url)
+        rhsm_output_fragment += 'Enabled: {0}\n'.format(enabled_repository.enabled)
+        rhsm_output_fragment += '\n'
+        rhsm_list_enabled_output += rhsm_output_fragment
+
+    external_command_runner.add_mocked_command_with_stdout(RHSM_LIST_ENABLED_REPOS, rhsm_list_enabled_output)
+
+    scanrhsm.scan()
+
+    assert actor_products.model_instances, 'The actor did not produce any messages.'
+
+    produced_message = actor_products.model_instances[0]
+
+    assert len(produced_message.enabled_repos) == 3, \
+        'The actor failed to detected enabled repositories on the system.'
+
+    for enabled_repository in RHSM_ENABLED_REPOS:
+        assert enabled_repository.id in produced_message.enabled_repos, \
+            'The actor failed to retrieve repository ID provided in the RHSM output.'
+
+
+def test_scansubscriptionmanagementinfo_get_existing_product_certificates(monkeypatch, common_monkeypatched_handles):
+    """Verifies that the actor is able to correctly retrieve existing product certificates."""
+    _original_isdir = os.path.isdir
+
+    def mocked_isdir(path):
+        # Only return True for these two specific paths specific to this test
+        if path == '/etc/pki/product-default':
+            return True
+        if path == '/etc/pki/product':
+            return False
+        return _original_isdir(path)
```

Comment:
for others `_original_XX` - unit-tests should not work with the real system. The environment should be fully mocked... so unit-tests do not touch anything real in the environment...

Reply:
Valuable lesson learned :+1:. The newest version throws an exception when the module attempts to communicate with (file)system in a way that it is not supposed/predicted to.